solidago/src/solidago/primitives.py

```diff
@@ -85,18 +85,23 @@ def _qr_quantile_loss_derivative(
     """Computes the derivative of the loss associated to qr_quantile"""
     regularization = (variable - default_value) / lipschitz
 
-    if quantile == 0.5:
-        quantile_term = 0.0
-    elif isinstance(voting_rights, (int, float)):
-        quantile_term = (1.0 - 2.0 * quantile) * voting_rights * len(values)
-    else:
-        quantile_term = (1.0 - 2.0 * quantile) * np.sum(voting_rights)
-
     deltas = variable - values
     uncertainties_2 = left_uncertainties_2 * (deltas < 0) + right_uncertainties_2 * (deltas > 0) + spacing
     forces = voting_rights * deltas / np.sqrt(uncertainties_2 + deltas**2)
+    
+    if quantile == 0.5:
+        return regularization + forces.sum()
+    
+    left_strength = min(1.0, quantile / (1-quantile))
+    right_strength = min(1.0, (1-quantile) / quantile)
+    
+    forces = np.where(
+        forces < 0,
+        forces * left_strength,
+        forces * right_strength,
+    )
 
-    return regularization + quantile_term + forces.sum()
+    return regularization + forces.sum()
```

Comment:
@lenhoanglnh This change seems to change significantly the behaviour of the "zero shift" on current Tournesol data. Is it expected? Should we adjust the quantile parameter?

On "main", after applying the shift with `score_shift_quantile = 0.15`, about 13% of the individual scores are negative. On this branch "neurips24", that would be 37%. 

As a consequence the distribution of Tournesol would be modified, with fewer videos reaching the recommendability threshold (1238 instead of 3013).

(I used the "legacy2023" pipeline, currently deployed on production. But I expect it would similar with the new pipeline).

![image](https://github.com/tournesol-app/tournesol/assets/4726554/c15e4783-fa22-40b3-8056-2e57e4a5cf8b)

Reply:
This is unsatisfactory indeed. 
I'm a bit disturbed. It feels like the quantile is now poorly estimated.
Maybe this is because videos with lower scores have higher uncertainty? Or less trust?

Reply:
OK I looked at the data and indeed, the uncertainties for bad videos are smaller than for good videos, which explains why the quantile increased with the new quantile definition. I see two simple fixes:
- Reduce ```score_shift_quantile = 0.15``` to ```score_shift_quantile = 0.05```.
- Remove uncertainties in quantile estimation.
The former is much more satisfactory.